docker-official-images/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"text/template"
+	"os"
+	"github.com/SonarSource/docker-sonarqube/docker-official-images/internal/build"
+	"github.com/SonarSource/docker-sonarqube/docker-official-images/internal/config"
+)
+
+// main is the entry point of the CLI application.
+func main() {
+	configFilePath := "active_versions.json"
+
+	// 1. Read and parse the activeVersions JSON file
+	activeConfigs, err := config.ParseConfigFile(configFilePath)
+	if err != nil {
+		log.Fatalf("Error reading or parsing active versions config: %v", err)
+	}
+
+    gitFetcher := build.NewGitFetcher() // Operates on current Git repo
+
+	// 3. Process each activeVersion to retrieve ImageBuildMetadata
+	var allBuildMetadata []build.ImageBuildMetadata
+	for _, cfg := range activeConfigs {
+
+		metadataList, err := build.GetBuildMetadataFromConfig(cfg, gitFetcher)
+		if err != nil {
+			log.Fatalf("Error processing active config for branch %q: %v", cfg.Branch, err)
+		}
+		allBuildMetadata = append(allBuildMetadata, metadataList...)
+	}
+
+	fmt.Printf("Successfully processed %d image build metadata entries.\n", len(allBuildMetadata))
+	// fmt.Println("--- Generated Build Metadata ---")
+	// for _, md := range allBuildMetadata {
+	// 	fmt.Printf("Branch: %s, Edition: %s, Version: %s, Directory: %s, Tags: %v, GitCommit: %s\n",
+	// 		md.Branch, md.EditionType, md.Version, md.ImageDirectory, md.Tags, md.GitCommit)
+	// }
+
+	templateFilePath := "official_images.tmpl"
```

Comment:
Can we split this `main` into multiple functions so it is easier to test?

Reply:
i dont think it make sense, this main function is like a CI script, it compile other tested functions.